pkg/resources/resource_grant_role.go

```diff
@@ -85,9 +85,10 @@ func grantRoleRead(ctx context.Context, d *schema.ResourceData, meta interface{}
 	// Check if role contains member
 	mapping, _ := materialize.ParseRolePrivileges(roles)
 
+	// Check if role contains member and if not, set id to empty string so that it can be recreated
 	if !slices.Contains(mapping[key.roleId], key.memberId) {
 		d.SetId("")
-		return diag.Errorf("role does contain member %s", key.memberId)
+		return nil
 	}
```

Comment:
There was an issue with the logical here when handling the situation where a role does not contain a member. The code checks whether the specified member is part of the role, and if not, it sets the resource ID to an empty string `(d.SetId(""))` which effectively tells Terraform that the resource does not exist. 

However, instead of just setting the ID to an empty string and returning `nil` which would indicate a successful "non-existence" handling to Terraform, the old code returned an error after setting the ID to an empty string which essentially never removed the non-existing resource from the state.